frontend/src/pages/SessionForm/pages/participants.js

```diff
@@ -54,6 +61,50 @@ const Participants = ({ formData }) => {
     fetchRecipients();
   }, [recipientOptions, regionId]);
 
+  // Get Groups.
+  const [groups, setGroups] = useState([]);
+  const [useGroups, setUseGroups] = useState(false);
+  const [groupRecipientIds, setGroupRecipientIds] = useState([]);
+  const [showGroupInfo, setShowGroupInfo] = useState(false);
+  const [fetchedGroups, setFetchedGroups] = useState(false);
+  useEffect(() => {
+    async function fetchGroups() {
+      if (regionId && !fetchedGroups) {
+        const retrievedGroups = await getGroupsForSession(regionId);
+
+        // Add recipientIds to groups.
+        const groupsWithRecipientIds = retrievedGroups.map((group) => ({
+          ...group,
+          // Match groups to grants as recipients could have multiple grants.
+          recipients: group.grants.map((g) => g.id),
+        }));
+        setGroups(groupsWithRecipientIds);
+        setFetchedGroups(true);
+      }
+    }
+    fetchGroups();
+  }, [regionId, groups, fetchedGroups]);
```

Comment:
![two fetches](https://github.com/HHS/Head-Start-TTADP/assets/3288586/b142b449-0aaa-499d-80f1-e2b4e5e7bef3)

This is running twice

Reply:
Probably swapping these would do it:

```js
setGroups(groupsWithRecipientIds);
setFetchedGroups(true);

Reply:
fixed by moving the set state before the call.